docker/runtime/bash/http-trigger/kubeless-shell.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+
+from subprocess import call
+import bottle
+import os
+import prometheus_client as prom
+
+app = application = bottle.app()
+
+cmdline = '/kubeless/' + os.getenv('MOD_NAME') + '.sh'
+cmdlineprep = 'chmod -R 755 /kubeless'
+
+def func(osparam):
+    global cmdline
+    if osparam:
+	cmdline = cmdline + ' ' + osparam
+    call(cmdlineprep, shell=True)
+    call(cmdline, shell=True)
+
+func_calls = prom.Counter('function_calls_total',
+                           'Number of calls to user function',
+                          ['method'])
+func_errors = prom.Counter('function_failures_total',
+                           'Number of exceptions in user function',
+                           ['method'])
+func_hist = prom.Histogram('function_duration_seconds',
+                           'Duration of user function in seconds',
+                           ['method'])
+
+
+@app.route('/', method=['GET', 'POST'])
+def handler():
+    req = bottle.request
+    method = req.method
+    func_calls.labels(method).inc()
+    print str(bottle.request)
+    with func_errors.labels(method).count_exceptions():
+        with func_hist.labels(method).time():
+            if method == 'GET':
+                return func(None)
+            else:
+                return func(bottle.request.body.read())
```

Comment:
Another alternative is to pass the request body to the shell command on stdin, and capture stdout as the reply.  Would this be "more shell-y"?

Reply:
I think it is fine to use arguments since we are calling functions

Reply:
Note that we still need to capture a result for http triggers (which is awkward with anything other than stdout from the shell command), and we won't be able to pass binary request data (with embedded `\0`) via argv args (eg: someone uploading a photo).

Reply:
> Note that we still need to capture a result for http triggers (which is awkward with anything other than stdout from the shell command)

Yeah, if the function requires a response it should return the value using stdout

> and we won't be able to pass binary request data

is still possible if the data is encoded in base64, isn't it?